packages/state/src/useControlledState.ts

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+import { useCallbackRef } from '@manifest-ui/hooks';
+
+export interface UseControlledStateProps<T> {
```

Comment:
Love this!! ❤️  I've always recoded this in components, but this makes too much sense!